drivers/i2c/i2c_ll_stm32.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2016 BayLibre, SAS
+ * Copyright (c) 2017 Linaro Ltd
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+#ifndef _STM32_I2C_H_
+#define _STM32_I2C_H_
+
+typedef void (*irq_config_func_t)(struct device *port);
+
+struct i2c_stm32_config {
+#ifdef CONFIG_I2C_STM32_INTERRUPT
+	irq_config_func_t irq_config_func;
+#endif
+	struct stm32_pclken pclken;
+	I2C_TypeDef *i2c;
+};
+
+struct i2c_stm32_data {
+#ifdef CONFIG_I2C_STM32_INTERRUPT
+	struct k_sem device_sync_sem;
+#endif
+	union dev_config dev_config;
+#ifdef CONFIG_I2C_STM32_V1
+	u16_t slave_address;
+	struct {
+		unsigned int is_restart;
+		unsigned int flags;
+#else
+	struct {
+#endif
```

Comment:
This is awkward to read, how about:

#ifdef CONFIG_I2C_STM32_V1
        u16_t slave_address;
#endif
        struct {
#ifdef CONFIG_I2C_STM32_V1
                unsigned int is_restart;
                unsigned int flags;
#endif
                unsigned int is_write;
...